src/main/kotlin/Todo/ToDoList.kt

```diff
@@ -0,0 +1,34 @@
+package Todo
+
+import Utilities.FileInterface
+import Utilities.JsonGenerator
+import java.io.FileNotFoundException
+
+class ToDoList(private val filePath: String, private val fileIo: FileInterface, private val jsonGenerator: JsonGenerator) {
+
+    fun addTask(task: Task): String {
+        val taskId = calculateId()
+        task.setId(taskId)
+        val jsonTask = jsonGenerator.resourceToJson(task)
+        fileIo.writeResource(filePath, jsonTask)
+        return task.body
+    }
+
+    fun retrieveList(): List<String> {
+        return try {
+            val tasks = fileIo.readResource(filePath)
+            tasks.split("\n").toList()
+        }catch (e: FileNotFoundException) {
+            listOf()
+        }
+    }
```

Comment:
Looks like you're reading the whole file. Do you think this could be an issue in the future when you have a lot of todos?